### PR TITLE
JBPM-9659: Upgrade deployments to WildFly 23

### PR DESCRIPTION
**Pull request by jesuino**
**JIRA**: 

[JBPM-9659](https://issues.redhat.com/browse/JBPM-9659)

**Related PRs*: 

https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/1693
https://github.com/kiegroup/kie-soup/pull/216
https://github.com/kiegroup/droolsjbpm-knowledge/pull/543
https://github.com/kiegroup/drools/pull/3757
https://github.com/kiegroup/optaplanner/pull/1463
https://github.com/kiegroup/lienzo-tests/pull/124
https://github.com/kiegroup/appformer/pull/1163
https://github.com/kiegroup/kie-uberfire-extensions/pull/122
https://github.com/kiegroup/jbpm/pull/1974
https://github.com/kiegroup/kie-jpmml-integration/pull/62
https://github.com/kiegroup/droolsjbpm-integration/pull/2539
https://github.com/kiegroup/kie-wb-playground/pull/115
https://github.com/kiegroup/kie-wb-common/pull/3660
https://github.com/kiegroup/drools-wb/pull/1506
https://github.com/kiegroup/jbpm-designer/pull/913
https://github.com/kiegroup/jbpm-work-items/pull/212
https://github.com/kiegroup/jbpm-wb/pull/1506
https://github.com/kiegroup/optaplanner-wb/pull/399
https://github.com/kiegroup/kie-wb-distributions/pull/1115
https://github.com/kiegroup/openshift-drools-hacep/pull/116
https://github.com/kiegroup/process-migration-service/pull/5
https://github.com/kiegroup/optaweb-employee-rostering/pull/662
https://github.com/kiegroup/optaweb-vehicle-routing/pull/535

### CODE REVIEW

File: kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/marshalling/json/FallbackableTypeFactory.java [hunk 1]
```diff
@@ -23,6 +23,7 @@
 import com.fasterxml.jackson.databind.util.ArrayBuilders;
 import com.fasterxml.jackson.databind.util.ClassUtil;
 import com.fasterxml.jackson.databind.util.LRUMap;
```
Comment (FireBurn) [inline kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/marshalling/json/FallbackableTypeFactory.java:25]: Shouldn't this line be removed now?

File: kie-server-parent/kie-server-tests/pom.xml [hunk 2]
```diff
@@ -944,6 +913,14 @@
...
           <artifactId>netty-transport-native-epoll</artifactId>
         </dependency>
+          <dependency>
+            <groupId>io.netty</groupId>
+            <artifactId>netty-codec-socks</artifactId>
+          </dependency>
+          <dependency>
+            <groupId>io.netty</groupId>
+            <artifactId>netty-handler-proxy</artifactId>
+          </dependency>
```
Comment (jesuino) [inline kie-server-parent/kie-server-tests/pom.xml:923]: Hello @lucamolteni See these deps we had to add for Kie Server JMS tests - these deps may also be required with KS JMS client (or netty-all)

File: kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/marshalling/json/FallbackableTypeFactory.java (hunk 1 above)
Comment (tkobayas) [reply kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/marshalling/json/FallbackableTypeFactory.java:25]: Hi @jesuino  , you can use this diff for FallbackableTypeFactory to align with jackson 2.12. (Note that this was not a PR to kiegroup but I wrote this to mbiarnes' repo when he was working on the wildfly update)

https://github.com/mbiarnes/droolsjbpm-integration/pull/8/files#diff-a5cbc3c1d03d90fca3a485f7bf685cee697c05878c17dea8969e19f9b1175c61

Of course, I can apply it later after you finish the current big PR. Thanks!

Comment (jesuino) [reply kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/marshalling/json/FallbackableTypeFactory.java:25]: Hello @tkobayas 

Sorry I didn't see your changes! We did faced it and solved it in a slightly different way in the revapi plugin.

When trying to apply the diff I had conflicts, would you please review the changes from this PR and possibly send a PR to my branch?

Thanks!

Comment (tkobayas) [reply kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/marshalling/json/FallbackableTypeFactory.java:25]: @jesuino Sent a PR : https://github.com/jesuino/droolsjbpm-integration/pull/12 Thanks!

**Comment by mareknovotny** [comment]
we should exclude here ignored tests in pom.xml plugin configuration to be able to run them with maven command property.

**Review by lucamolteni — APPROVED** [review]

**Comment by lucamolteni** [comment]
> we should exclude here ignored tests in pom.xml plugin configuration to be able to run them with maven command property.

Definitely - from my understanding we're only missing the Camel related one right?

**Comment by tiagobento** [comment]
As agreed, changes on how we ignore tests will be done as part of https://issues.redhat.com/browse/AF-2885 and https://issues.redhat.com/browse/AF-2886.

**Comment by tiagobento** [comment]
@lucamolteni Thanks for the approval, however, can you please take a look at the comment @jesuino left there regarding the JMS Client?

File: kie-server-parent/kie-server-tests/pom.xml (hunk 2 above)
Comment (lucamolteni) [reply kie-server-parent/kie-server-tests/pom.xml]: Ok Thanks @jesuino. Are the JMS tests failing? Or are they running without these

**Review by tomasdavidorg — APPROVED** [review]

**Pull request merged**